### PR TITLE
26 0 version of bitcoin

### PR DESCRIPTION
**Pull request by nolim1t**
Release 26-0 version of bitcoind containers

### CODE REVIEW

**Comment by nolim1t** [comment]
Looks like the sha256sums do not match.

```bash
% grep "bitcoin-26.0.tar.gz" SHA256SUMS      
ab1d99276e28db62d1d9f3901e85ac358d7f1ebcb942d348a9c4e46f0fcdc0a1  bitcoin-26.0.tar.gz

# From releases page https://github.com/bitcoin/bitcoin/releases/tag/v26.0
% sha256sum bitcoin-26.0.tar.gz 
3eb7c1e35de9415f82737e48cdc6f0e7f4b75973cb123996662c1650f6d544cf  bitcoin-26.0.tar.gz
```

**Comment by litln** [comment]
From https://github.com/bitcoin/bitcoin/releases/tag/v26.0
`Do not use the links provided by GitHub, rather use the above download links, they are guaranteed to be generated deterministically and signed.`

Change the source download url to https://bitcoincore.org/bin/bitcoin-core-26.0/ and it works.

File: 26.0/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+ARG VERSION
+
+# Download sigs
+ADD https://bitcoincore.org/bin/bitcoin-core-$VERSION/SHA256SUMS.asc ./
+# Download checksums
+ADD https://bitcoincore.org/bin/bitcoin-core-$VERSION/SHA256SUMS ./
+
+# Download source code from same website as github is probably deterministicly built
+ADD https://bitcoincore.org/bin/bitcoin-core-26.0/bitcoin-$VERSION.tar.gz ./bitcoin-$VERSION.tar.gz
```
Comment (litln) [inline 26.0/Dockerfile:63]: This still has the version 26.0 hardcoded

File: 26.0/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+# VERSION of Bitcoin Core to be build
+#   NOTE: Unlike our other images this one is NOT prefixed with `v`,
+#           as many things (like download URLs) use this form instead.
+ARG VERSION=26.0
+
+# CPU architecture to build binaries for
+ARG ARCH
+
+# Define default versions so that they don't have to be repeated throughout the file
+ARG VER_ALPINE=3.18
```
Comment (litln) [inline 26.0/Dockerfile:12]: bump this to 3.19 while you're at it?

File: 26.0/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+FROM preparer-base AS preparer-release
+
+ARG VERSION
+
+# Download sigs
+ADD https://bitcoincore.org/bin/bitcoin-core-$VERSION/SHA256SUMS.asc ./
+# Download checksums
+ADD https://bitcoincore.org/bin/bitcoin-core-$VERSION/SHA256SUMS ./
+
+# Download source code from same website as github is probably deterministicly built
```
Comment (fanquake) [inline 26.0/Dockerfile:62]: I think this comment needs fixing. GitHub might be deterministic, but it doesn't match our source tarballs, which are also deterministic. This comment seems to imply that our tarballs are not deterministic?

**Pull request merged**